assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js

```diff
@@ -121,26 +126,42 @@ function getVideoUrls(m) {
   // The rest of the application expects 2 named "sizes": "full", and "preview"
   // The highest fidelity is used (videoUrls is ordered by fidelity from the backend)
   // as "full", and the lowest as "preview".
-  if (m.imageUrls?.length < 2) {
+  if (m.videoUrls?.length < 2) {
     throw new Error('Invalid number of urls for asset. Need at least 2: ' + m);
   }
 
-  const sizesFromBiggest = m.videoUrls
-    .sort((x, y) => y.width - x.width)
-    .map((u) => mediaUrlToImageSizeDescription(m, u));
+  const sortedVideos = m.videoUrls.sort(
+    (x, y) => (y.width ?? 0) - (x.width ?? 0)
+  );
+  const originalSize = getOriginalSize(sortedVideos);
+  const sizesFromBiggest = sortedVideos.map((u) =>
+    mediaUrlToImageSizeDescription(m, u, originalSize)
+  );
+
   return {
     full: sizesFromBiggest[0],
     preview: sizesFromBiggest[sizesFromBiggest.length - 1],
   };
 }
 
-function mediaUrlToImageSizeDescription(media, url) {
+function getOriginalSize(mediaUrls) {
+  return {
+    originalWidth: mediaUrls[0].width,
+    originalHeight: mediaUrls[0].height,
+  };
+}
+
+function mediaUrlToImageSizeDescription(media, url, originalSize) {
+  const { originalWidth, originalHeight } = originalSize;
+  if (!originalWidth || !originalHeight) {
+    throw new Error('No original size present.');
+  }
   return {
     file: media.name,
     source_url: url.url,
     mime_type: url.mimeType,
-    width: url.width,
-    height: url.height,
+    width: url.width ?? 640,
```

Comment:
Hard coding of values is specific to coverr, right? 'media' already has the provider type, so it should be easy to add a precondition.

Reply:
Done.